src/jit/block.cpp

```diff
@@ -1374,6 +1374,56 @@ BasicBlock* Compiler::bbNewBasicBlock(BBjumpKinds jumpKind)
     return block;
 }
 
+//------------------------------------------------------------------------
+// blockhasEHBoundaryIn: Determine if this block begins at an EH boundary.
+//
+// Return Value:
+//    True iff the block is the target of an EH edge; false otherwise.
+//
+bool BasicBlock::hasEHBoundaryIn()
```

Comment:
This function will only detect if this is the first block of a handler. Do you care about whether it is the first block of a `try` region?

Reply:
Should you be using `BasicBlock::sameEHRegion()` somehow instead? Or would it make sense to assert something like (pseudo-code):
```
if (hasEHBoundaryIn())
{
 foreach pred of block:
    assert(!sameEHRegion(pred, block));
}
```

Reply:
I think that `BasicBlock::sameEHRegion()` might be the way to go. I'll look into it.

On the other - I wasn't sure if it was possible for there to be same-region edges as well as non-same-region edges. For register allocation, it only matters whether there's one such edge. Is it possible for there to be a block that is the first block of a handler, but perhaps also is targeted by a back-edge in the hander? Or would that cause a "loop-header" type block to be created to be the entry block of the handler?

> Do you care about whether it is the first block of a try region?

No, I don't think so, because if we have "normal" flow into a try region, I don't think we need to have all the live lclVars on the stack because that's an edge that's reflected in the flowgraph. Does that seem right?